crates/wallet/src/wallet/coin_selection.rs

```diff
@@ -52,11 +52,10 @@
 //!                 (&mut selected_amount, &mut additional_weight),
 //!                 |(selected_amount, additional_weight), weighted_utxo| {
 //!                     **selected_amount += weighted_utxo.utxo.txout().value.to_sat();
-//!                     **additional_weight += Weight::from_wu(
-//!                         (TxIn::default().segwit_weight().to_wu()
-//!                             + weighted_utxo.satisfaction_weight as u64)
-//!                             as u64,
-//!                     );
+//!                     **additional_weight += TxIn::default()
+//!                         .segwit_weight()
+//!                         .checked_add(weighted_utxo.satisfaction_weight)
+//!                         .expect("valid `Weight`'s");
```

Comment:
Note the expectation here is not just that the individual weights are valid but that summing them won't cause an overflow, maybe the expect message should clarify that.

But I guess the choice to use `checked_add` instead of `+` depends on whether we decide to use `Weight::MAX` in the case `max_weight_to_satisfy` returns an error. thoughts?

Reply:
> Note the expectation here is not just that the individual weights are valid but that summing them won't cause an overflow, maybe the expect message should clarify that.

Sure, I can improve the expectation message.

> But I guess the choice to use checked_add instead of + depends on whether we decide to use Weight::MAX in the case max_weight_to_satisfy returns an error. thoughts?

I don't think it strictly depends on that, but it's already as a safe/prevention for any sum that might cause overflow, in that scenario (use Weight::MAX) ends up it'll already be covered 🤔

Reply:
Ok let's stick with `checked_add` and make the expect say something like "integer overflow occurred"

Reply:
I updated it to: "`Weight` addition should not cause an integer overflow", trying to follow the [common styles/conventions](https://doc.rust-lang.org/std/error/index.html#common-message-styles).

Let me know what you think.